sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -572,17 +569,17 @@ export class KeysClient {
    * in the response. This operation requires the keys/list permission.
    * @param name Name of the key to fetch versions for
    * @param [options] The optional parameters
-   * @returns PagedAsyncIterableIterator<KeyAttributes>
+   * @returns PagedAsyncIterableIterator<KeyAttributes, KeyAttributes[]>
    */
   public listKeyVersions(
     name: string,
     options?: ListKeysOptions
-  ): PagedAsyncIterableIterator<KeyAttributes> {
+  ): PagedAsyncIterableIterator<KeyAttributes, KeyAttributes[]> {
     const iter = this.listKeyVersionsAll(name, options);
     return {
       async next() {
-        const item = (await iter.next()).value;
-        return item ? { done: false, value: item } : { done: true, value: undefined };
+        const item = await iter.next();
```

Comment:
Can you just `return iter.next()` ?

Reply:
Would it be `return await iter.next()` or `return iter.next()`?

Reply:
since it's an async function, and there is no local error handling for rejected promises, return aPromise and return await aPromise are identical.

Reply:
TIL

Reply:
Thanks, didn't see that !!